docs/source/architecture/slashing.rst

```diff
@@ -0,0 +1,72 @@
+The Slashing Protocol
+=====================
+
+TBD
+
+
+Violations
+----------
+
+TBD
+
+
+Calculating the slashing penalty
+--------------------------------
+
+TBD (https://github.com/nucypher/nucypher/issues/803)
+
+
+How slashing affects stake
+--------------------------
+
+The punishment is to reduce the number of tokens that belongs to the offending staker.
+In this case, the main task is not to violate the logic of locking tokens.
```

Comment:
Another meaning - when we slash Ursula we also should not break logic of sub stakes. If we will not decrease amount of sub stake then Ursula will get reward more than she deserve.

Reply:
```suggestion
However, in doing so, the protocol must not violate the logic governing the lock periods associated with a stake. 
```

Reply:
I hear emphasis on duration, but only amount of sub stakes are adjusted. Maybe I'm wrong

Reply:
I think, I didn't really understand what this sentence is trying to say. Neither in @szotov, nor in @arjunhassard variant

Reply:
This part of explanation is important, so need to rephrase before merge. The idea is from slashing perspective sub stakes are not hold information about how many tokens belongs to a stake. Sub stakes define when staker will get tokens back but also how big will be reward. Bigger sub stakes does not mean that staker has more tokens. So if we decrease amount of belonging tokens from 1000 to 900 and sum of sub stakes is still 1000 - then miner will get more reward for some periods. As a result decreasing sum of sub stakes to amount of tokens will be sufficient. This what I meant - do not breake logic of locking tokens

Reply:
Ok, this is how I understand (for others to have the same understanding too):
"Only part of the tokens held in the mining contract are locked in sub-stakes (others are available to withdraw: they could be from uncollected staking compensation). When slashing is happening, we first decrease the total amount of tokens held by the contract, and then adjust the sub-stakes so that they hold no less tokens than Ursula has in the contract. So, if misbehavior occurs, Ursula firstly loses her liquid (non-withdrawn) tokens, and then the tokens locked for staking"

Reply:
@michwill if your interpretation is correct it seems counterintuitive that you can lose non-staked tokens...

Reply:
That's my understanding based on conversation with @szotov. And that's why I translated it here for discussion: it may indeed seem counter-intuitive!
So what do we want? Lose only staked tokens, or anything you may have sitting in that smart contract?

Reply:
Non-staked tokens is unlocked sub stakes plus reward.
Of course if it's wrong way and should slash only locked tokens - I fix this in code and docs

Reply:
@michwill, @mswilkison I'm waiting your thoughts on that

Reply:
I suppose if someone begins producing malicious re-encryptions and they have unclaimed rewards, there isn't anything wrong with slashing those rewards even though it's counter-intuitive. It just provides an incentive not leave rewards sitting there and to either withdraw them or restake.

Reply:
Yeah, I think, I'd be ok with it, too